numbat/modules/core/lists.nbt

```diff
@@ -114,10 +114,39 @@ fn sort_by_key<A, D: Dim>(key: Fn[(A) -> D], xs: List<A>) -> List<A> =
                   sort_by_key(key, drop(floor(len(xs) / 2), xs)),
                   key)
 
-@description("Sort a list of quantities")
+@description("Sort a list of quantities in ascending order")
 @example("sort([3, 2, 7, 8, -4, 0, -5])")
 fn sort<D: Dim>(xs: List<D>) -> List<D> = sort_by_key(id, xs)
 
+fn _negate<D: Dim>(x: D) = -x
+
+@description("Sort a list of quantities in descending order")
+@example("sort_descending([3, 2, 7, 8, -4, 0, -5])")
+fn sort_descending<D: Dim>(xs: List<D>) -> List<D> = sort_by_key(_negate, xs)
+
+@description("looks for an element in a list. Returns true if the element is in the list.")
+@example("[3, 2, 7, 8, -4, 0, -5] |> contains(0)")
+@example("[3, 2, 7, 8, -4, 0, -5] |> contains(1)")
+fn contains<A>(x: A, xs: List<A>) -> Bool = 
+  if is_empty(xs)
+    then false
+    else if x == head(xs)
+      then true
+      else contains(x, tail(xs))
+
+fn _unique<A>(acc: List<A>, xs: List<A>) -> List<A> = 
+  if is_empty(xs)
+    then acc
+    else if is_empty(acc)
+      then _unique([head(xs)], tail(xs))
+      else if (acc |> contains(head(xs)))
+        then _unique(acc, tail(xs))
+        else _unique((cons_end(head(xs), acc)), tail(xs))
+
+@description("Remove duplicates, ensuring every value is unique")
```

Comment:
```suggestion
@description("Remove duplicates from a given list.")
```

Reply:
Does it only remove duplicates which are consecutive?

Reply:
No, it should return a list in which every value of the original list occurs exactly once. So no matter the order of elements this should hold.

Reply:
Just saw the tests, thanks.